src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx

```diff
@@ -3,223 +3,218 @@
+  const {
+    fields: dependencies,
+    append: appendDependency,
+    remove: removeDependency
+  } = useFieldArray({ control, name: 'dependencies' });
 
-  const deliverablesUtil: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = dels.slice();
-      clone.push({
-        id: -1,
-        detail: val,
-        isResolved: false
-      });
-      setDels(clone);
-    },
-    remove: (idx) => {
-      const clone = dels.slice();
-      clone.splice(idx, 1);
-      setDels(clone);
-    },
-    update: (idx, val) => {
-      const clone = dels.slice();
-      clone[idx].detail = val;
-      setDels(clone);
-    }
-  };
+  const { mutateAsync } = useEditWorkPackage(workPackage.wbsNum);
 
-  const setters = {
-    setProjectLead,
-    setProjectManager,
-    setName,
-    setCrId,
-    setStartDate,
-    setDuration,
-    setDeps,
-    setEa,
-    setDels,
-    setStatus
-  };
+  if (allUsers.isLoading || !allUsers.data || !auth.user) return <LoadingIndicator />;
+  if (allUsers.isError) {
+    return <ErrorPage message={allUsers.error?.message} />;
+  }
+
+  const users = allUsers.data.filter((u) => u.role !== 'GUEST');
 
   const transformDate = (date: Date) => {
     const month = date.getUTCMonth() + 1 < 10 ? `0${date.getUTCMonth() + 1}` : (date.getUTCMonth() + 1).toString();
     const day = date.getUTCDate() < 10 ? `0${date.getUTCDate()}` : date.getUTCDate().toString();
     return `${date.getUTCFullYear().toString()}-${month}-${day}`;
   };
 
-  const transformWbsNum = (wbsNum: WbsNumber) => {
-    return {
-      carNumber: wbsNum.carNumber,
-      projectNumber: wbsNum.projectNumber,
-      workPackageNumber: wbsNum.workPackageNumber
-    };
-  };
-
-  const handleSubmit = async (event: SyntheticEvent) => {
-    event.preventDefault();
-
-    const { userId } = auth.user!;
-
-    const payload = {
-      projectLead,
-      projectManager,
-      workPackageId: workPackage.id,
-      userId,
-      name,
-      crId: parseInt(crId),
-      startDate: transformDate(startDate),
-      duration,
-      dependencies: deps.map((dep) => transformWbsNum(dep)),
-      expectedActivities: ea,
-      deliverables: dels,
-      wbsElementStatus: status
-    };
-
-    console.log(payload);
+  const onSubmit = async (data: any) => {
+    const { name, projectLeadId, projectManagerId, startDate, duration, wbsElementStatus, crId, dependencies } = data;
+    const expectedActivities = mapBulletsToPayload(data.expectedActivities);
+    const deliverables = mapBulletsToPayload(data.deliverables);
 
     try {
+      const payload = {
+        projectLeadId,
+        projectManagerId,
+        workPackageId: workPackage.id,
+        userId,
+        name,
+        crId: parseInt(crId),
+        startDate: transformDate(startDate),
+        duration,
+        dependencies: dependencies.map((dep: any) => {
+          return validateWBS(dep.wbsNum);
+        }),
+        expectedActivities: expectedActivities,
```

Comment:
for these three you can just do `expectedActivieis, deliverables, wbsElementStatus }`